robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -107,15 +118,15 @@ def execute(self, userdata):
             return 'failed'
 
         # Make sure the torso and the arm are done
-        self.robot.torso.wait_for_motion_done(cancel=True)
+        #self.robot.torso.wait_for_motion_done(cancel=True)
         arm.wait_for_motion_done(cancel=True)
 
         # This is needed because the head is not entirely still when the
         # look_at_point function finishes
         rospy.sleep(rospy.Duration(0.5))
 
         # Update the entity (position)
-        segm_res = self.robot.ed.update_kinect("%s" % grab_entity.id)
+        # segm_res = self.robot.ed.update_kinect("%s" % grab_entity.id)
```

Comment:
With Hero you don't want to update the entity after going to the pre-grasp pose, so this part was moved to before the pre-grasp.